cpp/arcticdb/version/version_core.cpp

```diff
@@ -821,16 +821,26 @@ void read_incompletes_to_pipeline(
     pipeline_context->total_rows_ = pipeline_context->calc_rows();
 }
 
-void check_incompletes_index_ranges_dont_overlap(const std::shared_ptr<PipelineContext>& pipeline_context) {
-    // Does nothing if the symbol is not timestamp-indexed
-    // Checks both that the index ranges of incomplete segments do not overlap with one another, and that the earliest
-    // timestamp in an incomplete segment is greater than the latest timestamp existing in the symbol in the case of a
-    // parallel append
+void check_incompletes_index_ranges_dont_overlap(const std::shared_ptr<PipelineContext>& pipeline_context,
+                                                 const std::optional<SortedValue>& previous_sorted_value) {
+    /*
+     Does nothing if the symbol is not timestamp-indexed
+     Checks:
+      - that the existing data is not known to be unsorted in the case of a parallel append
+      - that the index ranges of incomplete segments do not overlap with one another
```

Comment:
Why is this an issue? AFAIK we allow repeating index values, so in theory we can gather all incomplete segments sort them and write them to storage. Am I missing something?